libs/core/include/core/random.hpp

```diff
@@ -27,5 +29,45 @@ struct Random
   static LaggedFibonacciGenerator<> generator;
 };
 
+
+/**
+ * Fisher-Yates shuffle algorithm for lfg
+ * @tparam T
+ */
+template <typename T>
+static void Shuffle(LaggedFibonacciGenerator<> & gen, std::vector<T> const &in_vec, std::vector<T> &out_vec)
+{
+  out_vec = in_vec;
+
+  for (std::size_t i = out_vec.size() - 1; i > 0; --i)
+  {
+    std::size_t j = gen() % (i + 1);
```

Comment:
I would shift this by 19. LCG is known to be alternating on first bit and subsequent bits to be strongly correlated.
```suggestion
    std::size_t j = (gen() >> 19) % (i + 1);
```